deepchecks/tabular/dataset.py

```diff
@@ -1061,6 +1065,172 @@ def datasets_share_date(cls, *datasets: 'Dataset') -> bool:
 
         return True
 
+    @property
+    def _dataset_description(self) -> t.Tuple[
```

Comment:
I'd pass here `max_rows_to_show` and in the start filter the data, since if it is a few millions in length, all dataframe operations here are expensive and un-necessary

Reply:
as much as I know, pandas does not touch data during slicing/column selection
it operates only over indexes therefore it is so effective

but, yup, it is better to change this
will fix

Reply:
> pandas does not touch data during slicing/column selection

right but when you do reset_index which adds new columns, it does copy the data

Reply:
fixed